pallets/messages/src/lib.rs

```diff
@@ -178,6 +183,9 @@ pub mod pallet {
 		) -> DispatchResultWithPostInfo {
 			let provider_key = ensure_signed(origin)?;
 
+			let schema = T::SchemaProvider::get_schema_by_id(schema_id);
+			ensure!(schema.is_some(), Error::<T>::InvalidSchemaId);
+
```

Comment:
maybe move this one under the other ensure so that it does not read if the second one fails.